cli/cmd/metrics.go

```diff
@@ -160,17 +160,9 @@ func getMetrics(
 		return nil, err
 	}
 
-	defer portforward.Stop()
-
-	go func() {
-		err := portforward.Run()
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Error running port-forward: %s", err)
-			portforward.Stop()
-		}
-	}()
-
-	<-portforward.Ready()
+	if err = portforward.Init(); err != nil {
```

Comment:
why do we no longer need to call `Stop` here?

Reply:
`Stop` there was to finish the port-forwarding in case of error. Now that's handled inside `Init`. The new comments in the upcoming push should make it clearer.

Reply:
if I'm reading the old code correctly, it looks like this was closing the port-forward after all bytes were finished reading from the response below.  Don't we still want to close the port-forward after we're finished reading?

Reply:
I think `Stop` is still needed here. Otherwise, `pf.stopCh` will not be closed, implying that the connection opened by `pf.Run()` (specifically, [`fw.ForwardPorts()`](https://godoc.org/k8s.io/client-go/tools/portforward#PortForwarder.ForwardPorts)) won't be closed. `cli/cmd/dashoard.go` still calls `Stop` on `SIGINT`.

Reply:
You're right, I was leaking `stopCh`. Will add an explicit call to `Stop()` on the calling side.